cmd/resources/resources.go

```diff
@@ -97,41 +113,81 @@ func (op *OperationCmd) initFlags() error {
 	return nil
 }
 
-func (op *OperationCmd) makeRequest(cmd *cobra.Command, args []string) error {
-	paramVals := map[string]interface{}{}
+func formatURL(baseURI, path string, urlParams []string) string {
+	s := make([]interface{}, len(urlParams))
+	for i, v := range urlParams {
+		s[i] = v
+	}
+
+	re := regexp.MustCompile(`{\w+}`)
+	format := re.ReplaceAllString(path, "%s")
+
+	return baseURI + fmt.Sprintf(format, s...)
+}
 
+func (op *OperationCmd) makeRequest(cmd *cobra.Command, args []string) error {
+	var data interface{}
 	if op.RequiresBody {
-		var data interface{}
 		// TODO: why does viper.GetString(cliflags.DataFlag) not work?
 		err := json.Unmarshal([]byte(cmd.Flags().Lookup(cliflags.DataFlag).Value.String()), &data)
 		if err != nil {
 			return err
 		}
-		paramVals[cliflags.DataFlag] = data
+	}
+	jsonData, err := json.Marshal(data)
+	if err != nil {
+		return err
 	}
 
+	query := url.Values{}
+	var urlParms []string
 	for _, p := range op.Params {
-		var val interface{}
-		switch p.Type {
-		case "string":
-			val = viper.GetString(p.Name)
-		case "boolean":
-			val = viper.GetBool(p.Name)
-		case "int":
-			val = viper.GetInt(p.Name)
```

Comment:
i think it's actually ok to just grab all of these as string since we're just passing them into the query param or url anyway